sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_repository.py

```diff
@@ -0,0 +1,228 @@
+from typing import TYPE_CHECKING
+
+from azure.core.exceptions import (
+    ClientAuthenticationError,
+    ResourceNotFoundError,
+    ResourceExistsError,
+    HttpResponseError,
+    map_error,
+)
+from azure.core.paging import ItemPaged
+from azure.core.pipeline import Pipeline
+from azure.core.tracing.decorator import distributed_trace
+
+from ._base_client import ContainerRegistryBaseClient, TransportWrapper
+from ._generated.models import AcrErrors
+from ._helpers import _parse_next_link
+from ._models import (
+    DeleteRepositoryResult,
+    ArtifactManifestProperties,
+    RepositoryProperties,
+)
+from ._registry_artifact import RegistryArtifact
+
+if TYPE_CHECKING:
+    from typing import Any, Dict
+    from azure.core.credentials import TokenCredential
+    from ._models import ContentProperties
+
+
+class ContainerRepository(ContainerRegistryBaseClient):
+    def __init__(self, endpoint, name, credential, **kwargs):
+        # type: (str, str, TokenCredential, Dict[str, Any]) -> None
+        """Create a ContainerRepository from an endpoint, repository name, and credential
+
+        :param str endpoint: An ACR endpoint
+        :param str name: The name of a repository
+        :param credential: The credential with which to authenticate
+        :type credential: :class:`~azure.core.credentials.TokenCredential`
+        :returns: None
+        :raises: None
+        """
+        if not endpoint.startswith("https://") and not endpoint.startswith("http://"):
+            endpoint = "https://" + endpoint
+        self._endpoint = endpoint
+        self.name = name
+        self._credential = credential
+        super(ContainerRepository, self).__init__(endpoint=self._endpoint, credential=credential, **kwargs)
+
+    @distributed_trace
+    def delete(self, **kwargs):
+        # type: (Dict[str, Any]) -> DeleteRepositoryResult
+        """Delete a repository
+
+        :returns: Object containing information about the deleted repository
+        :rtype: :class:`~azure.containerregistry.DeleteRepositoryResult`
+        :raises: :class:`~azure.core.exceptions.ResourceNotFoundError`
+        """
+        return DeleteRepositoryResult._from_generated(  # pylint: disable=protected-access
+            self._client.container_registry.delete_repository(self.name, **kwargs)
+        )
+
+    @distributed_trace
+    def get_properties(self, **kwargs):
+        # type: (Dict[str, Any]) -> RepositoryProperties
+        """Get the properties of a repository
+
+        :returns: :class:`~azure.containerregistry.RepositoryProperties`
+        :raises: :class:`~azure.core.exceptions.ResourceNotFoundError`
+        """
+        return RepositoryProperties._from_generated(  # pylint: disable=protected-access
+            self._client.container_registry.get_properties(self.name, **kwargs)
+        )
+
+    @distributed_trace
+    def list_manifests(self, **kwargs):
+        # type: (Dict[str, Any]) -> ItemPaged[ArtifactManifestProperties]
```

Comment:
Are the listed items here the same data that I would get from a `get_manifest_properties` call?
We're using the same response object for both - so just checking that they both include the same information.

Reply:
Yes they are, both calls return the same data.